pkg/controller/admissionchecks/provisioning/provisioning.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	autoscaling "k8s.io/autoscaler/cluster-autoscaler/apis/provisioningrequest/autoscaling.x-k8s.io/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/constants"
+	"sigs.k8s.io/kueue/pkg/util/admissioncheck"
+)
+
+func isProvisioned(pr *autoscaling.ProvisioningRequest) bool {
```

Comment:
I'm hesitant here, it feels this move of functions could be a dedicated cleanup PR. It distracts from reviewing the main body of changes. OTOH doing a little bit of drive-by cleanup should be fine.

Reply:
Since I've already addressed some comments regarding those function I'd rather stick with this PR for better visibility of changes if you don't mind. However, I'll keep this in mind and will dedicate a separate PR for this kind of changes in the future

Reply:
sgtm